### ISSUE TITLE
Use tax class of child product when product is configurable

### ISSUE THREAD

**Issue by crankycyclops** [taxjar/taxjar-magento2-extension#49]
I have a configurable product that has both downloadable and physical options (choice between an e-book and a print book.) Right now, the taxjar module determines whether or not sales tax should be applied based on the tax class of the parent configurable product, not the selected child product. This results in sales tax being applied to the e-book, even though in my state sales tax is not collected on digital goods.

Instead of determining taxability based on the parent product, the taxjar module should use the selected child product to determine whether or not sales tax applies.

Thank you!

**Comment by ryanreeves-taxjar** [comment]
Thanks @crankycyclops!  We'll take a look into this and get back to you soon.

**Comment by crankycyclops** [comment]
Hi, I was just wondering if there's been an update on this. I've already had customers that were erroneously charged sales tax in CA, and for now I've had to manually refund them for the difference. Thanks!

**Comment by crankycyclops** [comment]
I've been doing some digging into this myself, and have found a solution that works for me: https://magento.stackexchange.com/a/243665/73086

The issue has to do with how Magento determines the tax class of a configurable product during checkout. Once I implemented a custom module with the aforementioned plugin, the tax class is determined, in the case of a configurable product, but the selected child product.

I'm not sure if this solution belongs in your Taxjar module or not, since it addresses a fundamentally Magento issue that's independent of Taxjar's API, but it's trivial to implement if you guys decide to do so.

In the meantime, I've got this working for my store now, so it's no longer an issue.
